common/src/main/java/org/valkyrienskies/mod/mixin/entity/MixinEntity.java

```diff
@@ -125,13 +128,64 @@ private void preGetEyePosition(final float partialTicks, final CallbackInfoRetur
         cir.setReturnValue(newEyePos);
     }
 
+    /**
+     * @reason Needed for players to pick blocks correctly when mounted to a ship
+     */
+    @Inject(method = "getEyePosition()Lnet/minecraft/world/phys/Vec3;", at = @At("HEAD"), cancellable = true)
+    private void preGetEyePositionServer(final CallbackInfoReturnable<Vec3> cir) {
+        final ShipMountedToData shipMountedToData = VSGameUtilsKt.getShipMountedToData(Entity.class.cast(this), null);
```

Comment:
add comment: why do we need to use the dragging information here? why doesn't it automatically work having replaced the position of the player with the correct position?

Reply:
Explained via comment.